addons/web/controllers/main.py

```diff
@@ -451,6 +423,178 @@ def _admin_password_warn(uid):
+
+        :param: string; addon, the addon the template is declared in
+        :param: etree node: the current template we are are handling
+        :returns (str, str, str)
+        """
+        full_parent_name = template.attrib[self.STATIC_INHERIT_DIRECTIVE]
+        split_name_attempt = full_parent_name.split('.', 1)
+        src_addon, src_template = tuple(split_name_attempt) if len(split_name_attempt) == 2 else (addon, full_parent_name)
+        if src_addon not in self.template_dict:
+            if full_parent_name in self.template_dict[addon]:
+                src_addon = addon
+                src_template = full_parent_name
+            else:
+                raise ValueError(_('Module %s not loaded or inexistent, or templates of addon being loaded (%s) are misordered') % (src_addon, addon))
+
+        if src_template not in self.template_dict[src_addon]:
+            raise ValueError(_("No template found to inherit from. Module %s and template name %s") % (src_addon, src_template))
+
+        return full_parent_name, src_addon, src_template
+
+    def _compute_xml_tree(self, addon, file_name, source):
+        """Computes the xml tree that 'source' contains
+        Applies inheritance specs in the process
+
+        :param string: addon: the current addon we are reading files for
+        :param string: file_name: the current name of the file we are reading
+        :param string: source: the content of the file
+        :returns etree
+        """
+        try:
+            tree = etree.parse(io.BytesIO(source), parser=etree.XMLParser(remove_comments=True)).getroot()
+        except etree.ParseError as e:
+            _logger.error("Could not parse file %s: %s" % (file_name, e.msg))
+            raise e
+
+        self.template_dict.setdefault(addon, OrderedDict())
+        for child in list(tree):
+            if self.NAME_TEMPLATE_DIRECTIVE in child.attrib:
+                name = child.attrib[self.NAME_TEMPLATE_DIRECTIVE]
+            else:
+                name = 'anonymous_template_%s' % len(self.template_dict[addon])
+            if self.STATIC_INHERIT_DIRECTIVE in child.attrib:
+                src_mode = child.attrib.get(self.STATIC_INHERIT_MODE_DIRECTIVE, self.DEFAULT_MODE)
+                if src_mode not in self.STATIC_INHERIT_MODES:
+                    raise ValueError(_("Invalid inherit mode value"))
+
+                full_parent_name, src_addon, src_template = self._get_parent_coordinates(addon, child)
+
+                # After several performance tests, we found out that deepcopy is the most efficient
+                # solution in this case (compared with copy, xpath with '.' and stringifying).
+                parent = copy.deepcopy(self.template_dict[src_addon][src_template])
+                parent_tag = parent.tag
+                parent.tag = 't'
+                xpaths = list(child)
+                if self.debug and src_mode == self.STATIC_INHERIT_MODES[1]:
```

Comment:
self.STATIC_INHERIT_MODES[1] does not read well. I could rather read EXTENSION_KEYWORD